db/db_impl/db_impl_open.cc

```diff
@@ -445,8 +451,11 @@ Status DBImpl::Recover(
   if (!immutable_db_options_.best_efforts_recovery) {
     s = versions_->Recover(column_families, read_only, &db_id_);
   } else {
+    if (read_only) {  // TODO: get rid of this case?
```

Comment:
Move the comment to a new line. This is not a simple note, and more text may be expected, thus moving to new line so that adding new text is easier.

Reply:
We can avoid this to make code cleaner.
Pass both `fs` and `db_file_names` to `ManifestPicker`. If `db_file_names` is empty, then use `fs` to list directory.